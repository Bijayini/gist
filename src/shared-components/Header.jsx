import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';

import routePaths from '../lib/routePaths';
import theme from '../lib/theme';

import BodyCopy from '../ui-components/BodyCopy';

import labels from './labels/Header.lables';

const Header = () => (
  <Fragment>
    <header className="header">
      <Link to={routePaths.HOME}>
        <BodyCopy
          variant="extra-bold"
          xs="x-large"
          xm="x-large"
          xl="x-large"
          color="white"
        >
          {labels.github}
          &nbsp;
        </BodyCopy>
        <BodyCopy
          variant="regular"
          xs="x-large"
          xm="x-large"
          xl="x-large"
          color="white"
        >
          {labels.gist}
        </BodyCopy>
      </Link>
    </header>
    <style jsx="true">
      {`
        .header {
          background: ${theme.primaryColor};
          padding: 20px;
        }
      `}
    </style>
  </Fragment>
);
export default Header;
