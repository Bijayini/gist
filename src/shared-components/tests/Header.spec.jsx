import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

describe('<Header />', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Header />);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});
