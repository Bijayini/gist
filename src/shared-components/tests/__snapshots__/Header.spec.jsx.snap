// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match with snapshot 1`] = `
"<Fragment>
  <header className=\\"jsx-3197027717 header\\">
    <Link to=\\"/\\" replace={false}>
      <BodyCopy variant=\\"extra-bold\\" xs=\\"x-large\\" xm=\\"x-large\\" xl=\\"x-large\\" color=\\"white\\" className=\\"\\" uppercase={false} styles={false}>
        GitHub
        Â 
      </BodyCopy>
      <BodyCopy variant=\\"regular\\" xs=\\"x-large\\" xm=\\"x-large\\" xl=\\"x-large\\" color=\\"white\\" className=\\"\\" uppercase={false} styles={false}>
        Gist
      </BodyCopy>
    </Link>
  </header>
  <JSXStyle id=\\"1560694114\\" dynamic={{...}}>
    .header.__jsx-style-dynamic-selector{background:#cc1c5d;padding:20px;}
  </JSXStyle>
</Fragment>"
`;
